scripts/generators/generator_utils.py

```diff
@@ -1,25 +1,148 @@
+            for key, value in vuid_dict.items():
+                if key == "vuid":
+                    yield value
+                elif isinstance(value, dict):
+                    for vuid in ExtractVUIDs(value):
+                        yield vuid
+                elif isinstance (value, list):
+                    for listValue in value:
+                        for vuid in ExtractVUIDs(listValue):
+                            yield vuid
+
+    valid_vuids = set()
+    if not os.path.isfile(valid_usage_file):
+        print(f'Error: Could not find, or error loading {valid_usage_file}')
+        sys.exit(1)
+    json_file = open(valid_usage_file, 'r', encoding='utf-8')
+    vuid_dict = json.load(json_file)
+    json_file.close()
+    if len(vuid_dict) == 0:
+        print(f'Error: Failed to load {valid_usage_file}')
+        sys.exit(1)
+    for json_vuid_string in ExtractVUIDs(vuid_dict):
+        valid_vuids.add(json_vuid_string)
+
+    # List of VUs that should exists, but have a spec bug
+    for vuid in [
+        # https://gitlab.khronos.org/vulkan/vulkan/-/issues/3582
+        "VUID-VkCopyImageToImageInfoEXT-commonparent",
+        "VUID-vkUpdateDescriptorSetWithTemplate-descriptorSet-parent",
+        "VUID-vkUpdateVideoSessionParametersKHR-videoSessionParameters-parent",
+        "VUID-vkDestroyVideoSessionParametersKHR-videoSessionParameters-parent",
+        "VUID-vkGetDescriptorSetHostMappingVALVE-descriptorSet-parent",
+        ]:
+        valid_vuids.add(vuid)
+
+    return valid_vuids
+
+# Will do a sanity check the VUID exists
+def getVUID(valid_vuids: set, vuid: str, quotes: bool = True) -> str:
+    if vuid not in valid_vuids:
+        print(f'Warning: Could not find {vuid} in validusage.json')
+        vuid = vuid.replace('VUID-', 'UNASSIGNED-')
+    return vuid if not quotes else f'"{vuid}"'
+
 class PlatformGuardHelper():
     """Used to elide platform guards together, so redundant #endif then #ifdefs are removed
-    Note - be sure to call addGuard(None) when done to add a trailing #endif if needed
+    Note - be sure to call add_guard(None) when done to add a trailing #endif if needed
     """
     def __init__(self):
-        self.currentGuard = None
+        self.current_guard = None
 
-    def addGuard(self, guard):
+    def add_guard(self, guard, extra_newline = False):
         out = []
-        if self.currentGuard != guard:
-            if self.currentGuard != None:
-                out.append(f'#endif  // {self.currentGuard}\n')
-            if guard != None:
-                out.append(f'#ifdef {guard}\n')
-            self.currentGuard = guard
+        if self.current_guard is not guard and self.current_guard is not None:
+            out.append(f'#endif  // {self.current_guard}\n')
+        if extra_newline:
+            out.append('\n')
+        if self.current_guard is not guard and guard is not None:
+            out.append(f'#ifdef {guard}\n')
+        self.current_guard = guard
```

Comment:
```suggestion
        if self.current_guard != guard and self.current_guard is not None:
            out.append(f'#endif  // {self.current_guard}\n')
        if extra_newline:
            out.append('\n')
        if self.current_guard != guard and guard is not None:
            out.append(f'#ifdef {guard}\n')
        self.current_guard = guard
```
Apparently replacing `!=` with `is not` provokes different behavior.

Reply:
ok. that means VVL is broken too, because I just copied the 'newer' scripts over... We have script madness.

Reply:
should be fixed now. I'll push a fix to VVL in a sec

Reply:
https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/7787